netx/resolver/dnsoverhttps_test.go

```diff
@@ -92,7 +92,8 @@ func TestUnitDNSOverHTTPSSuccess(t *testing.T) {
 				},
 			}, nil
 		},
-		URL: "https://doh.powerdns.org/",
+		URL:          "https://doh.powerdns.org/",
+		HostOverride: "powerdns.org",
```

Comment:
```suggestion
		HostOverride: "",
```

I assume we don't need to set the override here?